openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/OpenShiftAssistant.java

```diff
@@ -160,4 +170,37 @@ public DeploymentConfig deploymentConfig() {
             .withName(this.applicationName)
             .get();
     }
+
+    public OpenShiftAssistant usingTemplate(String templateURL) {
```

Comment:
Instead of returning `OpenShiftAssistant` I would return a class called `OpenShiftAssistantTemplate` which receives as constructor the `client`, then this class would contain the method to add parameters and the `deploy` method.

In this way you are avoiding that someone mistakenly calls `deploy` method without setting any template.